WikiDrips/WikiRequest.swift

```diff
@@ -7,17 +7,67 @@
 //
 
 import Foundation
-import Accounts
-import Social
-import CoreLocation
 
 public class WikiRequest
 {
-    public func fetchWikiDoc(handler: @escaping ([WikiDoc]) -> Void) {
-//      TODO: make api requests
-            var wikiDocs = [WikiDoc]()
-            var wikiDocArray: NSArray?
-        
-
+    let session = URLSession.shared
+    let baseEndpoint: String = "http://en.wikipedia.org/w/api.php?action=query&list=search&srwhat=text&sroffset=0&format=json"
+    var wikiDocs = [[WikiDoc]]()
+    let searchText: String
+    
+    init?(searchText: String) {
+        self.searchText = searchText
+    }
+    
+    public func fetchWikiDocs(handler: @escaping ([[WikiDoc]]) -> Void) {
+        guard let urlRequest = urlRequest(searchText: searchText) else {
+            return
+        }
+        let task = session.dataTask(with: urlRequest) {
+            (data, response, error) in
+            // check for any errors
+            guard error == nil else {
+                print("error calling GET on \(self.baseEndpoint)")
+                print(error!)
```

Comment:
These errors are great for us (devs) but is the idea that whoever calls this is not aware that an error occurred?